static/app/components/searchQueryBuilder/hooks/useQueryBuilderState.tsx

```diff
@@ -685,15 +778,71 @@ export function useQueryBuilderState({
           });
         case 'DELETE_TOKENS':
           return deleteQueryTokens(state, action);
-        case 'UPDATE_FREE_TEXT':
-          return updateFreeText(state, action);
-        case 'REPLACE_TOKENS_WITH_TEXT':
-          return replaceTokensWithText(state, {
+        case 'UPDATE_FREE_TEXT': {
+          const {
+            query,
+            focusOverride,
+            committedQuery: currentCommittedQuery,
+          } = updateFreeText(state, action);
+
+          let replacedQuery: string | undefined;
+          if (
+            replaceRawSearchKeys &&
+            replaceRawSearchKeys.length > 0 &&
+            hasRawSearchReplacement
+          ) {
+            const newQuery = replaceFreeTextTokens(
+              action,
+              getFieldDefinition,
+              replaceRawSearchKeys,
+              query
+            );
+            replacedQuery = newQuery;
+          }
+
+          return {
+            focusOverride,
+            committedQuery: action.shouldCommitQuery
+              ? (replacedQuery ?? query)
+              : currentCommittedQuery,
+            query: replacedQuery ?? query,
+            replacedRawSearchKey: replacedQuery ? true : state.replacedRawSearchKey,
+          };
+        }
+        case 'REPLACE_TOKENS_WITH_TEXT': {
```

Comment:
This action gets used in a few different places so this can break other interactions like selecting and typing new text:

https://github.com/user-attachments/assets/2a803a81-99d8-49ee-a083-a7d8a3c58870